libsignal-service/src/push_service.rs

```diff
@@ -350,32 +350,85 @@ pub trait PushService {
         file: Option<(&str, &'s mut C)>,
     ) -> Result<(), ServiceError>;
 
+    fn build_verification_code_request_url(
+        msg_type: &str,
+        phone_number: &str,
+        captcha: Option<&str>,
+        challenge: Option<&str>,
```

Comment:
Doesn't it make more sense to have this in two/three separate methods, since only one of them can be `Some`, and none of these is actually dynamically dispatched?

Reply:
For the implementation here that would mean quite some code duplication, I think. We could get rid of the url formatting method though - I keep forgetting that we can assign complex if statements. For the account manager that could make more sense maybe.